ApiFirst/DomainServices/Services/CustomerServices.cs

```diff
@@ -0,0 +1,114 @@
+        }
+
+        public bool Delete(int id)
+        {
+            int index = _customers.FindIndex(customer => customer.Id == id);
+            if (index == -1) return false;
+            _customers.RemoveAt(index);
+            return true;
+        }
+
+        public List<Customer> GetAll()
+        {
+            return _customers;
+        }
+
+        public Customer? GetByCpf(string cpf)
+        {
+            cpf = cpf.Trim().Replace(".", "").Replace("-", "");
+            return _customers.FirstOrDefault(customer => customer.Cpf == cpf);
+        }
+
+        public int Update(int id, Customer model)
+        {
+
+            int index = _customers.FindIndex(customer => customer.Id == id);
+            if (index == -1) return -1;
+
+            if (ExistsUpdate(id, model)) return 0;
+            else
+            {
+                model.Id = _customers[index].Id;
+                _customers[index] = model;
+                return 1;
+            }
+        }
+
+        public Customer? GetById(int id)
+        {
+            return _customers.FirstOrDefault(x => x.Id == id);
+        }
+
+        public int Modify(int id, Customer model)
+        {
+            int index = _customers.FindIndex(customer => customer.Id == id);
+            if (index == -1) return -1;
+
+            if (ExistsUpdate(id, model)) return 0;
+            else
+            {
+                model.Id = _customers[index].Id;
+
+                if (_customers[index].FullName != model.FullName) _customers[index].FullName = model.FullName;
+
+                if (_customers[index].Email != model.Email) _customers[index].Email = model.Email;
+
+                if (_customers[index].EmailConfirmation != model.EmailConfirmation) _customers[index].EmailConfirmation = model.EmailConfirmation;
+
+                if (_customers[index].Cpf != model.Cpf) _customers[index].Cpf = model.Cpf;
+
+                if (_customers[index].Cellphone != model.Cellphone) _customers[index].Cellphone = model.Cellphone;
+
+                if (_customers[index].DateOfBirth != model.DateOfBirth) _customers[index].DateOfBirth = model.DateOfBirth;
+
+                if (_customers[index].EmailSms != model.EmailSms) _customers[index].EmailSms = model.EmailSms;
+
+                if (_customers[index].Whatsapp != model.Whatsapp) _customers[index].Whatsapp = model.Whatsapp;
+
+                if (_customers[index].Country != model.Country) _customers[index].Country = model.Country;
+
+                if (_customers[index].City != model.City) _customers[index].City = model.City;
+
+                if (_customers[index].PostalCode != model.PostalCode) _customers[index].PostalCode = model.PostalCode;
+
+                if (_customers[index].Address != model.Address) _customers[index].Address = model.Address;
+
+                if (_customers[index].Number != model.Number) _customers[index].Number = model.Number;
+
```

Comment:
Essas verificações não fazem sentido. Se a tua função pede um CustomerModel, você pode simplesmente substituir o objeto que está na lista pelo que foi passado como parâmetro
```suggestion
 _customer[index] = model;

```

Reply:
essa é minha tentativa de um Patch. Então, ele n pode substituir todo o model.

Reply:
> essa é minha tentativa de um Patch. Então, ele n pode substituir todo o model.

Então acredito que a lógica está errada. Se você passa um model inteiro e faz a comparação com o que está na lista, qualquer campo diferente será substituído pelo outro, o que, na prática, o torna um put. O patch seria passar como argumento apenas o valor do campo e, daí sim, fazer a atribuição.

Reply:
Entedido. Vou fazer este patch diferente para as versões que estão em lista. Quando passarmos para Database, fica mais simples.

Reply:
vou colocar para mudar o Email, como argumento